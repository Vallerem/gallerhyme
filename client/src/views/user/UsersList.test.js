import ConnectedUserList, { UsersList } from "./UsersList";

import { Provider } from "react-redux";
import configureStore from "../../redux/store";

const store = configureStore();

describe("<UserList/>", () => {
  describe("renders itsel", () => {
    it("without Redux", () => {
      const wrapper = shallow(<UsersList />);
      expect(wrapper).toMatchSnapshot();
    });
    it("with Redux", () => {
      const wrapper = shallow(<ConnectedUserList store={store} />);
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe("triggers", () => {
    it("componentDidMount method", () => {
      const spy = jest.spyOn(ConnectedUserList.prototype, "componentDidMount");
      const wrapper = shallow(<ConnectedUserList store={store} />).dive();
      expect(spy).toHaveBeenCalled();
    });
  });
});
