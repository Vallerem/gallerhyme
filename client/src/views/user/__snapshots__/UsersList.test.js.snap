// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserList/> renders itsel with Redux 1`] = `
<UsersList
  error={null}
  fetchAllUsers={[Function]}
  loading={false}
  page={1}
  pages={null}
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "unsubscribe": [Function],
    }
  }
  users={Array []}
/>
`;

exports[`<UserList/> renders itsel without Redux 1`] = `
<div>
  <h1>
    Users
  </h1>
  <hr />
  <InfiniteScroll
    dataLength={0}
    disableBrowserPullToRefresh={true}
    endMessage={
      <p
        style={
          Object {
            "textAlign": "center",
          }
        }
      >
        <b>
          No more items to see
           
          <span
            aria-label="eyes icon"
            role="img"
          >
            ðŸ‘€
          </span>
        </b>
      </p>
    }
    hasMore={true}
    loader={
      <h4>
        Loading...
      </h4>
    }
    next={[Function]}
    pullDownToRefreshContent={
      <h3>
        Pull down to refresh
      </h3>
    }
    pullDownToRefreshThreshold={100}
    releaseToRefreshContent={
      <h3>
        Release to refresh
      </h3>
    }
  />
</div>
`;
